### PR TITLE
docs: Create Samples for transfer manager

### PR DESCRIPTION
**Pull request by sydney-munro**
Samples for the upload many operations only. Will do chunked once Parallel Composite Upload is complete.

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/storage/transfermanager/DownloadBucket.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+class DownloadBucket {
+
+  public static void downloadBucketContents(String projectId,
+      String bucketName, Path destinationDirectory) {
+    Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+    List<BlobInfo> blobs = storage
+        .list(bucketName)
+        .streamAll()
+        .map(blob -> blob.asBlobInfo())
```
Comment (BenWhitehead) [inline samples/snippets/src/main/java/com/example/storage/transfermanager/DownloadBucket.java:39]: Nit
```suggestion
        .map(Blob::asBlobInfo)
```

File: samples/snippets/src/main/java/com/example/storage/transfermanager/UploadDirectory.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+class UploadDirectory {
+
+  public static void uploadDirectoryContents(String bucketName, Path sourceDirectory)
```
Comment (BenWhitehead) [inline samples/snippets/src/main/java/com/example/storage/transfermanager/UploadDirectory.java:33]: This needs a test

File: samples/snippets/src/main/java/com/example/storage/transfermanager/DownloadBucket.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import com.google.cloud.storage.transfermanager.ParallelDownloadConfig;
+import com.google.cloud.storage.transfermanager.TransferManager;
+import com.google.cloud.storage.transfermanager.TransferManagerConfig;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+
+class DownloadBucket {
+
+  public static void downloadBucketContents(String projectId,
```
Comment (BenWhitehead) [inline samples/snippets/src/main/java/com/example/storage/transfermanager/DownloadBucket.java:33]: This also needs a test

**Review by BenWhitehead — CHANGES_REQUESTED** [review]

**Review by BenWhitehead — APPROVED** [review]
lgtm from java side, probably want to get andrew to review for the general review.

**Review by andrewsg — APPROVED** [review]
LGTM

**Pull request merged**